include/nbl/ext/LumaMeter/CLumaMeter.h

```diff
@@ -118,13 +130,67 @@ class CLumaMeter : public core::TotalInterface
 			E_METERING_MODE meterMode, float minLuma=1.f/2048.f, float maxLuma=65536.f
 		);
 
-		// we expect user binds correct pipeline, descriptor sets and pushes the push constants by themselves
-		static inline void dispatchHelper(video::IVideoDriver* driver, const DispatchInfo_t& dispatchInfo, bool issueDefaultBarrier=true)
+		template <E_METERING_MODE mode>
+		static inline void updateDescriptorSet(
+			video::IGPUDescriptorSet* ds,
+			const core::smart_refctd_ptr<video::IGPUBuffer> paramUbo,
+			const asset::SBufferRange<video::IGPUBuffer>& outputSSBORange,
+			const core::smart_refctd_ptr<video::IGPUImageView> inputImageView,
+			video::ILogicalDevice* logicalDevice)
 		{
-			driver->dispatch(dispatchInfo.workGroupCount[0], dispatchInfo.workGroupCount[1], dispatchInfo.workGroupCount[2]);
+			static_assert(mode < EMM_COUNT, "Invalide meter mode!");
+
+			video::IGPUDescriptorSet::SWriteDescriptorSet writes[DEFAULT_DESCRIPTOR_COUNT] = {};
+			video::IGPUDescriptorSet::SDescriptorInfo infos[DEFAULT_DESCRIPTOR_COUNT] = {};
+
+			const asset::E_DESCRIPTOR_TYPE descriptorTypes[DEFAULT_DESCRIPTOR_COUNT] =
+			{
+				asset::EDT_UNIFORM_BUFFER, // luma input params
+				asset::EDT_STORAGE_BUFFER, // luma output buffer
+				asset::EDT_COMBINED_IMAGE_SAMPLER // input image
+			};
+
+			infos[0].desc = paramUbo;
+			infos[0].buffer.offset = 0ull;
+			infos[0].buffer.size = sizeof(Uniforms_t<mode>);
+
+			infos[1].desc = outputSSBORange.buffer;
+			infos[1].buffer.offset = outputSSBORange.offset;
+			infos[1].buffer.size = outputSSBORange.size - infos[1].buffer.offset;
```

Comment:
this function can go bye bye, no default descriptors and runtime GLSL re-binding of the image descriptor etc. 

User will provide an Image Accessor as a template param and stateful accessor